lib/src/se/llbit/chunky/ui/SizeInput.java

```diff
@@ -0,0 +1,300 @@
+        " (",
+        Bindings.createIntegerBinding(
+          () -> (int) Math.ceil((double) getWidthValue() / gdc.get()),
+          gdc, widthValueProperty()
+        ),
+        " : ",
+        Bindings.createIntegerBinding(
+          () -> (int) Math.ceil((double) getHeightValue() / gdc.get()),
+          gdc, heightValueProperty()
+        ),
+        ")"
+      )
+    );
+    changeQueuedProperty().addListener((observable, oldValue, newValue) -> {
+      // size changes were committed
+      if(!newValue) {
+        gdc.set(QuickMath.gcd(getWidthValue(), getHeightValue()));
+      }
+    });
+    return details;
+  }
+
+  public IntegerProperty widthValueProperty() {
+    return currentWidthProperty;
+  }
+
+  public IntegerProperty heightValueProperty() {
+    return currentHeightProperty;
+  }
+
+  public IntegerProperty queuedWidthValueProperty() {
+    return widthInput.valueProperty();
+  }
+
+  public IntegerProperty queuedHeightValueProperty() {
+    return heightInput.valueProperty();
+  }
+
+  public BooleanProperty ratioLockedProperty() {
+    return ratioLockCheckBox.selectedProperty();
+  }
+
+  public StringProperty unitStringProperty() {
+    return unitLabel.textProperty();
+  }
+
+  public ReadOnlyBooleanProperty changeQueuedProperty() {
+    return changeQueuedProperty;
+  }
+
+  public BooleanProperty showAspectRatioDetailsProperty() {
+    return showAspectRatioDetailsProperty;
+  }
+
+  public boolean isRatioLocked() {
+    return ratioLockedProperty().get();
+  }
+
+  public int getWidthValue() {
+    return widthValueProperty().get();
+  }
+
+  public int getHeightValue() {
+    return heightValueProperty().get();
+  }
+
+  public int getQueuedWidthValue() {
+    return queuedWidthValueProperty().get();
+  }
+
+  public int getQueuedHeightValue() {
+    return queuedHeightValueProperty().get();
+  }
+
+  /**
+   * @param width > 0
+   * @param height > 0
+   */
+  public void setSize(int width, int height) {
+    if(width < 1 || height < 1)
+      throw new IllegalArgumentException("Size must be positive (w,h > 0)");
+    widthValueProperty().set(width);
+    heightValueProperty().set(height);
+    submitChanges();
+  }
+
+  public void scaleSize(double scale) {
+    setSize(
+      scaleAxis(getWidthValue(), scale),
+      scaleAxis(getHeightValue(), scale)
+    );
+  }
+
+  /**
+   * Submits queued changes to the size change listener.
+   */
+  public void submitChanges() {
+    if(widthInput.isValid()) widthValueProperty().set(getQueuedWidthValue());
+    if(heightInput.isValid()) heightValueProperty().set(getQueuedHeightValue());
+    sizeChangedCallback.accept(getWidthValue(), getHeightValue());
+    changeQueuedProperty.set(false);
+  }
+
+  /**
+   * Updates the current size change listener.
+   *
+   * @param onChangeCallback if null remove current listener, otherwise add/replace current listener
+   */
+  public void addSizeChangeListener(BiConsumer<Integer, Integer> onChangeCallback) {
```

Comment:
Javadoc (and behavior) contradict the method name here :upside_down_face: 

```suggestion
  public void setSizeChangeListener(BiConsumer<Integer, Integer> onChangeCallback) {
```